tsl/src/continuous_aggs/common.h

```diff
@@ -0,0 +1,148 @@
+									/* This should also be the column used by time_bucket */
+	Oid htpartcoltype;
+	int64 htpartcol_interval_len; /* interval length setting for primary partitioning column */
+	int64 bucket_width;			  /* bucket_width of time_bucket, stores BUCKET_WIDTH_VARIABLE for
+									 variable-sized buckets */
+	Oid bucket_width_type;		  /* type of bucket_width */
+	Interval *interval;			  /* stores the interval, NULL if not specified */
+	const char *timezone;		  /* the name of the timezone, NULL if not specified */
+
+	FuncExpr *bucket_func; /* function call expr of the bucketing function */
+	/*
+	 * Custom origin value stored as UTC timestamp.
+	 * If not specified, stores infinity.
+	 */
+	Timestamp origin;
+} CAggTimebucketInfo;
+
+typedef struct AggPartCxt
+{
+	struct MatTableColumnInfo *mattblinfo;
+	bool added_aggref_col;
+	/*
+	 * Set to true when you come across a Var
+	 * that is not inside an Aggref node.
+	 */
+	bool var_outside_of_aggref;
+	Oid ignore_aggoid;
+	int original_query_resno;
+	/*
+	 * "Original variables" are the Var nodes of the target list of the original
+	 * CREATE MATERIALIZED VIEW query. "Mapped variables" are the Var nodes of the materialization
+	 * table columns. The partialization query is the one that populates those columns. The
+	 * finalization query should use the "mapped variables" to populate the user view.
+	 */
+	List *orig_vars; /* List of Var nodes that have been mapped to materialization table columns */
+	List *mapped_vars; /* List of Var nodes of the corresponding materialization table columns */
+					   /* orig_vars and mapped_vars lists are mapped 1 to 1 */
+} AggPartCxt;
+
+/* Note that we set rowsecurity to false here. */
+#define CAGG_MAKEQUERY(selquery, srcquery)                                                         \
+	do                                                                                             \
+	{                                                                                              \
+		(selquery) = makeNode(Query);                                                              \
+		(selquery)->commandType = CMD_SELECT;                                                      \
+		(selquery)->querySource = (srcquery)->querySource;                                         \
+		(selquery)->queryId = (srcquery)->queryId;                                                 \
+		(selquery)->canSetTag = (srcquery)->canSetTag;                                             \
+		(selquery)->utilityStmt = copyObject((srcquery)->utilityStmt);                             \
+		(selquery)->resultRelation = 0;                                                            \
+		(selquery)->hasAggs = true;                                                                \
+		(selquery)->hasRowSecurity = false;                                                        \
+		(selquery)->rtable = NULL;                                                                 \
+	} while (0);
```

Comment:
Transform it into a static inline function like `makeContinuousAggregateQuery`

Reply:
I am keeping it as is for the moment, because it is being accessed in multiple files.